frontend-zs/src/components/stories/StoryPage.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from 'react'
+import { Button, Media, Container } from 'react-bootstrap'
+import Axios from 'axios'
+import { Link, useParams } from 'react-router-dom'
+import { TiArrowBack } from "react-icons/ti"
+import StoryDetails from './StoryDetails'
+import StoryPagination from './StoryPagination'
+import StoryQuestionLike from './StoryQuestionLike'
+import { authenticationService } from '../../services/authentication.service';
+import './Story.scss'
+
+function goBack() {
+    return (
+        <Button className="btn-back mb-3" size="sm" active>
+            <Link to="/stories" >
+                <Media>
+                    <TiArrowBack size={20} />
+                    <Media.Body>Back</Media.Body>
+                </Media>
+            </Link>
+        </Button>
+    )
+}
+
+function StoryText(displayStory) {
+    return (
+        <div className="p-4 new-line" style={{ fontSize: 19 }} >
+            {displayStory}
+        </div>
+    )
+}
+
+export default function StoryPage() {
+    const params = useParams();
+    const [storyData, setStoryData] = useState({
+        title: '',
+        author: '',
+        authorImage: '',
+        story: [],
+        categories: [],
+        numberLikes: []
+    })
+    const [displayStory, setDisplayStory] = useState('')
+    const [currentUserId, setCurrentUserId] = useState('')
+
+    useEffect(() => {
+        Axios.get(process.env.REACT_APP_HOST + ':' + process.env.REACT_APP_PORT + '/api/stories/' + params.id)
+            .then(response => {
+                setStoryData(response.data)
+            })
+            .catch(function (error) {
+                console.log(error.message)
+            });
+        const currentUser = authenticationService.currentUserValue;
+        setCurrentUserId(currentUser._id)
+    }, [])
+
+    function onPageChanged(currentPage) {
+        setDisplayStory(storyData.story[currentPage - 1])
+    }
+
+    function onLikeClicked(storyLiked) {
+        let likesToSave = []
+        if (storyLiked) {
+            likesToSave = storyData.numberLikes
```

Comment:
We could add a check here to make sure that the current user does not exist in the likes array.

Reply:
There is a check in StoryQuestionLike.js:
`useEffect(() => {
        setStoryLiked(numberLikes.includes(currentUserId))
    }, [numberLikes])`

Won't it be double checked if we also add one here?

Reply:
@ahmad-diaa What about this issue?

Reply:
okay, makes sense.